flexget/components/ftp/sftp_client.py

```diff
@@ -0,0 +1,423 @@
+                    tries -= 1
+                    retry_interval += RETRY_STEP
+
+        return sftp
+
+    def close(self) -> None:
+        """
+        Close the sftp connection
+        """
+        self._sftp.close()
+
+    def _put_file(self, source: str, destination: str) -> None:
+        return self._sftp.put(source, destination)
+
+    def _download_file(self, source: str, destination: str, delete_origin: bool) -> None:
+
+        dir_name: str = remote_path.dirname(source)
+        destination_path: str = self._get_destination_path(source, destination)
+        destination_dir: str = local_path.dirname(destination)
+
+        if local_path.exists(destination_path):
+            self._logger.verbose('Skipping {} because destination file {} already exists. ', source, destination)
+            return
+
+        if not local_path.exists(destination_dir):
+            os.makedirs(destination_dir)
+
+        self._logger.verbose('Downloading file {} to {}', source, destination)
+
+        try:
+            self._sftp.get(source, destination_path)
+        except Exception as e:
+            self._logger.error('Failed to download {} ({})', source, e)
+            if local_path.exists(destination_path):
+                self._logger.debug('Removing partially downloaded file {}', destination_path)
+                os.remove(destination_path)
+            raise e
+
+        if delete_origin:
+            self.remove_file(source)
+            self.remove_dir(dir_name)
+
+    def _get_prefix(self) -> str:
+        """
+        Generate SFTP URL prefix
+        """
+
+        def get_login_string() -> str:
+            if self.username and self.password:
+                return '%s:%s@' % (self.username, self.password)
+            elif self.username:
+                return '%s@' % self.username
+            else:
+                return ''
+
+        def get_port_string() -> str:
+            if self.port and self.port != 22:
+                return ':%d' % self.port
+            else:
+                return ''
+
+        return 'sftp://%s%s%s/' % (get_login_string(), self.host, get_port_string())
+
+    @staticmethod
+    def _get_destination_path(path: str, destination: str) -> str:
+        relative_path = local_path.join(*remote_path.split(path))  # convert remote path style to local style
+        return local_path.join(destination, relative_path)
+
+
+class SftpError(Exception):
+    pass
+
+
+class HandlerBuilder:
+    """
+    Class for building pysftp.Connection.walktree node handlers.
+
+    :param sftp: A Connection object
+    :param logger: a logger object
+    :param url_prefix: SFTP URL prefix
+    """
+
+    def __init__(self, sftp: 'pysftp.Connection', logger: logging.Logger, url_prefix: str):
+        self._sftp = sftp
+        self._logger = logger
+        self._prefix = url_prefix
+
+    def get_file_handler(self, get_size: bool, entry_accumulator: list) -> Callable[[str], None]:
```

Comment:
For `Callable` the input types go in a list as the first argument, and the return type as the second. So that's a function that accepts `str` and returns a `None` type.

Reply:
I stand corrected, I apologize. I still think using it as it's own variable would be nice though

Reply:
No problem. The whole point of this PR is to improve code quality, so it makes sense to be thorough. As someone who mostly works in statically-typed languages, being able to alias types like that is so weird!